services/blockchain-connector/shared/sdk/genesisBlock.js

```diff
@@ -0,0 +1,106 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const {
+	Logger,
+	Exceptions: { TimeoutException },
+} = require('lisk-service-framework');
+
+const { getNodeInfo } = require('./actions_1');
+const { getGenesisBlockFromFS } = require('./blocksUtils');
+const { timeoutMessage, invokeAction } = require('./client');
+const { decodeBlock } = require('./decoder');
+const { parseToJSONCompatObj } = require('../jsonTools');
+
+const logger = Logger();
+
+let genesisHeight;
+let genesisBlockID;
+
+const getGenesisHeight = async () => {
+	if (!genesisHeight) {
+		const nodeInfo = await getNodeInfo();
+		genesisHeight = nodeInfo.genesisHeight;
+	}
+	return genesisHeight;
+};
+
+const getGenesisBlock = async (includeAccounts = false) => {
+	try {
+		const block = await getGenesisBlockFromFS();
+		const { header: { asset: { accounts, ...remAsset }, ...remHeader }, payload } = block;
+		const finalAccounts = includeAccounts ? accounts : [];
+		const metaMessage = 'Fetch the genesis accounts with \'getNumberOfGenesisAccounts\' and \'getGenesisAccounts\' methods';
+
+		return {
+			header: {
+				asset: {
+					...remAsset,
+					accounts: finalAccounts,
+				},
+				...remHeader,
+			},
+			payload,
+			meta: {
+				isGenesisBlock: true,
+				message: includeAccounts ? '' : metaMessage,
+			},
+		};
+	} catch (_) {
+		logger.debug('Genesis block snapshot retrieval was not possible, attempting to retrieve directly from the node.');
+	}
+
+	const height = await getGenesisHeight();
+	try {
+		const encodedBlock = await invokeAction('app:getBlockByHeight', { height });
+		const block = await decodeBlock(encodedBlock);
+		return { ...parseToJSONCompatObj(block), _raw: encodedBlock };
+	} catch (err) {
+		if (err.message.includes(timeoutMessage)) {
+			throw new TimeoutException('Request timed out when calling \'getGenesisBlock\'');
+		}
+		throw err;
+	}
+};
+
+const getGenesisBlockID = async () => {
+	if (!genesisBlockID) {
+		// eslint-disable-next-line no-use-before-define
```

Comment:
I think this comment is unnecessary here.

Reply:
Directive removed!